### PR TITLE
Add test-canisters.sh script to CI

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Make sure that test-canisters.sh script is kept up-to-date in case something else is needed to run tests.

# Changes

Add a test in CI to run the tests with this script.

<hr/><details><summary>🟡 Some screens were changed</summary></details>

### CODE REVIEW

File: .github/workflows/canister-tests.yml [hunk 1]
```diff
@@ -391,6 +390,20 @@ jobs:
         env:
           RUST_BACKTRACE: 1
 
+  test-canisters-script:
```
Comment (frederikrothenberger) [inline .github/workflows/canister-tests.yml:393]: This job takes more than 10 minutes, so it is immediately the slowest thing in the whole pipeline. I would not add it without any caching.

For reference, the normal pipeline with caching takes ~6m 30s.

**Review by frederikrothenberger — COMMENTED** [review]
Could you please split the changes and the reformatting into different PRs?

File: .github/workflows/canister-tests.yml (hunk 1 above)
Comment (lmuntaner) [reply .github/workflows/canister-tests.yml:393]: I added the `--no-build` parameter and used the built wasms. It takes 6'30" now. Is this reasonable?

It depends on a few other jobs though. Therefore, the whole CI is probably slower now.

I could add pocket-ic and the other dependencies into a cache and use it as well. Maybe in another PR?

**Comment by lmuntaner** [comment]
@frederikrothenberger ready for another review.

The duration is now down to 2min.

File: .github/workflows/canister-tests.yml [hunk 2]
```diff
@@ -391,6 +390,49 @@ jobs:
...
+        run: mv internet_identity_test.wasm.gz internet_identity.wasm.gz
+
+      - name: Create dummy assets
+        run: |
+          mkdir dist
+          touch dist/index.html
+
+      - name: Run tests
+        run: |
+          ./scripts/test-canisters.sh --no-build should_keep_new_anchor_across_rollback
```
Comment (frederikrothenberger) [inline .github/workflows/canister-tests.yml:428]: Maybe add a comment on how you chose these particular tests?

**Review by frederikrothenberger — APPROVED** [review]
LGTM, thanks!

File: .github/workflows/canister-tests.yml (hunk 2 above)
Comment (lmuntaner) [reply .github/workflows/canister-tests.yml:434]: done

**Pull request merged**